src/core/math/blue-noise.js

```diff
@@ -0,0 +1,50 @@
+
+let data = null;
+const initData = () => {
+    if (!data) {
+        const binaryString = atob(base64String);
+
+        // the data stores an array of RGBA 8-bit values, each of the color channels represents
+        // a separate blue noise stream
+        data = Uint8Array.from(binaryString, char => char.charCodeAt(0));
+    }
+};
+
+const blueNoiseData = () => {
+    initData();
+    return data;
+};
+
+/**
+ * Blue noise based random numbers API.
+ *
+ * @ignore
+ */
+class BlueNoise {
+    seed = 0;
+
+    constructor(seed = 0) {
+        this.seed = seed * 4;
+        initData();
+    }
+
+    next() {
+        this.seed = (this.seed + 4) % data.length;
+    }
+
+    value() {
+        this.next();
+        return data[this.seed] / 255;
+    }
+
+    vec4(dest = new Vec4()) {
```

Comment:
This api seems weird. Why expose `next` if `value` and `vec4` both call it? Shouldn't vec4 be called `vec4Value` or similar?

Reply:
next is meant as a private function - it moves to a next pixel (so 4 indices). Only value and vec4 are public (we can add vec2 and vec3 later too)